File: services/mini-coupang/src/test/java/com/deepblog/minicoupang/domain/seller/storage/SellerRepositoryIntegrationTest.java
package com.deepblog.minicoupang.domain.seller.storage;

import com.deepblog.minicoupang.domain.seller.Seller;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
import org.testcontainers.containers.PostgreSQLContainer;
import org.testcontainers.junit.jupiter.Container;
import org.testcontainers.junit.jupiter.Testcontainers;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
@Testcontainers
class SellerRepositoryIntegrationTest {

    @Container
    @ServiceConnection
    static PostgreSQLContainer<?> postgres = new PostgreSQLContainer<>("postgres:16");

    @Autowired
    SellerRepository repository;

    @Test
    void save_and_find_seller_round_trip() {
        Seller saved = repository.save(Seller.create("Toss Seller", "toss@example.com"));

        assertThat(saved.getId()).isNotNull();

        Seller found = repository.findById(saved.getId()).orElseThrow();
        assertThat(found.getName()).isEqualTo("Toss Seller");
        assertThat(found.getEmail()).isEqualTo("toss@example.com");
        assertThat(found.getVersion()).isEqualTo(0L);
    }

    @Test
    void findByEmail_returns_empty_when_no_match() {
        assertThat(repository.findByEmail("no-such@example.com")).isEmpty();
    }
}

File: services/mini-coupang/src/test/kotlin/com/deepblog/minicoupang/domain/seller/storage/SellerRepositoryIntegrationTest.kt
package com.deepblog.minicoupang.domain.seller.storage

import com.deepblog.minicoupang.domain.seller.Seller
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.testcontainers.service.connection.ServiceConnection
import org.testcontainers.containers.PostgreSQLContainer
import org.testcontainers.junit.jupiter.Container
import org.testcontainers.junit.jupiter.Testcontainers

@SpringBootTest
@Testcontainers
class SellerRepositoryIntegrationTest {

    companion object {
        @Container
        @ServiceConnection
        @JvmStatic
        val postgres: PostgreSQLContainer<*> = PostgreSQLContainer("postgres:16")
    }

    @Autowired lateinit var repository: SellerRepository

    @Test
    fun `save and find seller round-trip`() {
        val saved = repository.save(
            Seller(name = "Toss Seller", email = "toss@example.com"),
        )

        assertThat(saved.id).isNotNull()

        val found = repository.findById(saved.id!!).orElseThrow()
        assertThat(found.name).isEqualTo("Toss Seller")
        assertThat(found.email).isEqualTo("toss@example.com")
        assertThat(found.version).isEqualTo(0L)
    }

    @Test
    fun `findByEmail returns null when no match`() {
        val found = repository.findByEmail("no-such@example.com")
        assertThat(found).isNull()
    }
}
